lib/components/ScheduleSelector/components/MonthlySelector/MonthlySelector.tsx

```diff
@@ -62,22 +70,27 @@ const MonthlySelector: FC<MonthlySelectorProps> = ({
 
   const handleDateChange = useCallback(
     (days: string) => {
-      onChange({ months: monthlyData.months, time, days })
+      const validDays = everyMonth
+        ? days
+            .split(',')
+            .filter((day) => parseInt(day, 10) <= 28)
+            .join(',')
+        : days
+      onChange({ months: monthlyData.months, time, days: validDays })
     },
-    [monthlyData.months, time, onChange]
+    [monthlyData.months, time, everyMonth, onChange]
   )
 
-  const daysOfMonth = useMemo(
-    () =>
-      Array.from({ length: 28 }, (_, i) => {
-        const day = (i + 1).toString()
-        return {
-          label: day,
-          value: day.padStart(2, ZERO_STRING)
-        }
-      }),
-    []
-  )
+  const daysOfMonth = useMemo(() => {
+    const maxDays = everyMonth ? 28 : 31
```

Comment:
Maybe i didn't understand, but what about specific 30 days months?

Reply:
The current decision is that for "every month" , we'll show up to 28 days (since you cannot say every month at day 30), for specific month , we'll show option to select up to 31, and if you're selecting months which the days are not in the range, you'll get an error from the BE. For example, you cannot select days: "30, 31" and months "Mar, Apr"

Reply:
And we don't know the select month in order to show the right number of days?
I guess it's an edge case, but we're avoiding backend error.

Reply:
Yes, we can do it, but it is adding a lot of complications in the UI (it is not just one month that can be selected, so for each selection you need to check if the months array include February, or Month that has 31, or month that has 30, and update the days selection accordingly, and what if the user already selected 31, and then trying to add February to the selection? we'll remove it? )
There are many more cases that we can think of, and @jacky has decided to leave it for now

Reply:
@gonenidan , I've worked on the dynamic UI, pushed those changes, I hope that I've covered all